pkgs/development/python-modules/typing-validation/default.nix

```
@@ -0,0 +1,53 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+
+, pytestCheckHook
+
+, pythonOlder
+
+, setuptools-scm
+
+, numpy
+, typing-extensions
+}:
+
+buildPythonPackage rec {
+  pname = "typing-validation";
+  version = "1.2.11";
+  pyproject = true;
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "hashberg-io";
+    repo = "typing-validation";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-0scXoAPkx/VBIbNRMtFoRRbmGpC2RzNRmQG4mRXSxrs=";
+  };
+
+  build-system = [
+    setuptools-scm
+  ];
+
+  dependencies = [
+    numpy
+    typing-extensions
```

```suggestion
  dependencies = lib.optionals (pythonOlder "3.11") [
    typing-extensions
```
Yes, this package can be used for numpy, but it is not a required dependency.
So that we don't add it to the dependencies.
https://github.com/hashberg-io/typing-validation/blob/26de9324d98aa908e6767973325bd8d73aa541bb/setup.cfg#L31-L32

---

> Yes, this package can be used for numpy, but it is not a required dependency. So that we don't add it to the dependencies. https://github.com/hashberg-io/typing-validation/blob/26de9324d98aa908e6767973325bd8d73aa541bb/setup.cfg#L31-L32

On python 3.12 module not working without typing-extensions....
`ModuleNotFoundError: No module named 'typing_extensions'`